src/util.js

```diff
@@ -312,44 +327,70 @@ function isPrimitive (data) {
 }
 
 export function searchDeepInObject (obj, searchTerm) {
-  var match = false
+  return internalSearchObject(obj, searchTerm.toLowerCase(), new Map(), 0)
+}
+
+function internalSearchObject (obj, searchTerm, seen, depth) {
+  let match = false
   const keys = Object.keys(obj)
+  let key, value
   for (let i = 0; i < keys.length; i++) {
-    const key = keys[i]
-    const value = obj[key]
-    if (compare(key, searchTerm) || compare(value, searchTerm)) {
-      match = true
+    key = keys[i]
+    value = obj[key]
+    match = interalSearchCheck(searchTerm, key, value, seen, depth + 1)
+    if (match) {
       break
     }
-    if (isPlainObject(value)) {
-      match = searchDeepInObject(value, searchTerm)
-      if (match) {
-        break
-      }
+  }
+  return match
+}
+
+function internalSearchArray (array, searchTerm, seen, depth) {
+  let match = false
+  let value
+  for (let i = 0; i < array.length; i++) {
+    value = array[i]
+    match = interalSearchCheck(searchTerm, null, value, seen, depth + 1)
+    if (match) {
+      break
     }
   }
   return match
 }
 
-function compare (mixedValue, stringValue) {
-  if (Array.isArray(mixedValue) && searchInArray(mixedValue, stringValue.toLowerCase())) {
-    return true
+function interalSearchCheck (searchTerm, key, value, seen, depth) {
+  if (depth > 10) {
```

Comment:
It's okay for a hotfix, but wouldn't it be better to use something like `WeakMap` to make sure we're not checking an object we have checked before?

Reply:
It's what the `seen` `Map` is for.

Reply:
oh lol sorry, I thought we couldn't use objects as keys in js maps, not sure why.
Why do we wait for depth to be 10, wouldn't be safe to stop if we check the same object twice while inspecting an object?

Reply:
For performance reasons, I though it would be nice to have some kind of limit.

Reply:
> I thought we couldn't use objects as keys in js maps

It's the other way around, we can use objects in `Map` but not in `WeakMap`.

Reply:
I've used WeakMap with objects before 😆 

About the depth, I don't get what you mean with performance, if we drop the search because we found an object we already found while looking recursively into anothe object, wouldn't that be the sortest way to stop the search?

```js
// given
{ a: { b: obj }, c: obj}
```
once we try to lookup `obj` inside of `b`, we would never look it up inside of `c`


```js
// given
var a = {}
var b = { b: { a }}
a.b = b
```
 we would only dive till `a.b.b` (or `a.b.b.a` depending on the implem)

Reply:
Okay, I don't why I had errors with `WeakMap`.
If you don't have any match, we could spend a lot of time on very large objects, I thing it's a good idea to have some kind of hard limit on recursive depth (also, the algorithm can search deeply into arrays now).

Reply:
Ah, okay so it was to limit deep nested objects no matter repetition.
Forget about WeakMap, Map is ok 🙂